packages/graphback-core/src/runtime/CRUDService.ts

```diff
@@ -166,7 +166,9 @@ export class CRUDService<Type = any> implements GraphbackCRUDService<Type>  {
 
   public batchLoadData(relationField: string, id: string | number, filter: any, context: GraphbackContext) {
     // TODO use relationfield mapping
-    const keyName = `${this.modelName}${upperCaseFirstChar(relationField)}DataLoader`;
+    const selectedFields = context.graphback.options?.selectedFields || [];
+    const fetchedKeys = selectedFields.join('-');
+    const keyName = `${this.modelName}-${upperCaseFirstChar(relationField)}-${fetchedKeys}-DataLoader`;
```

Comment:
I think filters should be part of the key too. 

What do you think? @craicoverflow @wtrocki